### ISSUE TITLE
Grafana-ui: SegmentAsync remote typeahead

### ISSUE THREAD

**Issue by seanlaff** [grafana/grafana#21005]
The contract for `SegementAsync` includes a `query` param however I don't think it's supplied anywhere. https://github.com/grafana/grafana/blob/7d6bd78e0b592bcb9e9904a3f222953655019903/packages/grafana-ui/src/components/Segment/SegmentAsync.tsx#L10

Would love to be able to use this for async-typeahead in the case where there are too many results to fetch for the dropdown.

**Comment by torkelo** [comment]
can you take a look at that?

**Comment by sunker** [comment]
Hi @seanlaff! That was the idea from the beginning - that SegmentAsync would support lazy async loading, but I never got time to implement that. The query param was apparently never removed. 

I don't have time to add async typeahead right now. If you have time, you can give it a go. If not, I'll do it in a few weeks.

**Comment by seanlaff** [comment]
Hey @sunker I tried implementing this today but it looks like react-select does not support lazy-loading _default_ options. 

Clicking on an async-select will render an empty select... and will not fetch anything until the user starts typing (I assume we'd want it to make an async request for `""`). React-select offers us the ability to supply `defaultOptions = true`, but it does the fetch on mount- not menuOpen.

I looked into hooking into the `onMenuOpen` callback, but the acrobatics get pretty crazy and there's not really a way to pass "controlled" options back down into the async select.

I see you guys are using a react-select fork from @torkelo - maybe it's something we can play with there and get upstreamed?

Here's an example I was playing with from their docs that shows the issue https://codesandbox.io/s/react-codesandboxer-example-xojvl
